### ISSUE TITLE
[2] Open-Set Semi-Supervised Object Detection

### ISSUE THREAD

**Issue by sy00n** [sy00n/DL_paper_review#2]
# Introduction
본 논문에서 제안하는 방법론은 Open-Set Semi-Supervised Object Detection task이다. 기존에 Semi-Supervised object detection 모델들은 unlabeled data(open-set 카테고리 포함)을 더 많이 추가해서 학습할수록 낮은 성능을 보임을 발견했는데, 이는 기본적으로 SSL에서 unlabeled data를 더 활용할수록 좋을 것이라는 belief와는 다른 결과이다. 이러한 현상을 semantic expansion issue라고 하며, OOD object가 high confidence를 가지고 in-distribution object로 잘못 예측하게 되고 이로 인해 pseudo-label로 잘못 사용하게 되서 성능 저하로 이어지게 되는 것이다. (아래 fig1 참고)
  
![image](https://user-images.githubusercontent.com/67910856/233130027-696fd4ba-d68b-41ab-9245-93efdb2bab31.png)

이렇게 OOD sample을 사용했을때 오히려 성능을 저하시키는 해로운 영향을 제거하기 위해 기존 SSOD 학습 파이프라인에 추가 OOD filtering process를 추가할 것을 제안한다. 구체적으로는 먼저 OOD filtering을 수행하기 위해 oneline OOD detector를 사용한다. 하지만 그냥 적용할 경우 bounding box localization, box classification과 같은 다른 task와의 간섭 때문에 만족스러운 결과를 보이지 못했다. 따라서 object detector와 분리된 offline detection module을 제안한다. 그리고 self-sup DINO 모델을 기반으로 하는데, 이것의 장점은 1) pre-training Dino는 label annotation을 필요로 하지 않아 low-level setting에 적절하고 2) Online OOD 방법과 비교했을 때 pseudo-label에서 OOD object를 탐지하는 데 더 효과적이며 이는 위에서 언급한 해로운 영향을 제거한다. 3) OOd detector와 object detector가 서로 독립적이기 때문에 각각 따로 학습되고 위에서 언급한 간섭을 방지할 수 있다. 

### Contribution
- 최초로 open-set semi-supervised object detection(OSSOD) task를 다루고 기존 SSOD 방법의 한계를 분석한다.
- OSSOD task에서의 문제점을 밝히고 online 및 offline OOD detector를 제시하며 이 두 모듈간의 체계적인 비교를 제공한다.
- 광범위한 실험을 진행함.

# Semantic Expansion in Open-Set scenarios
![image](https://user-images.githubusercontent.com/67910856/233135464-64d37a48-e60f-4450-87fb-c6e90ec4ab9a.png)
![image](https://user-images.githubusercontent.com/67910856/233135696-8a9f0919-246c-48f9-959f-5f43290e60fe.png)

fig3을 보면 Open-set 설정에서는 제한된 성능 향상 폭을 보임을 발견했고 이러한 원인이 semantic expansion(fig 2a) 때문이라고 주장한다. 구체적으로 말하면 OOD/novel object가 unlabeled set에 있으면, closed-set classifier(Faster-RCNN에서는 ROI head)가 high confidence를 가지고 ID object로 예측하게 되는 것이다. 이로 인해 Over-confident, incorrect OOD instance는 confidence threesholding 후에도 ID object로 분류되는 경향이 있으며 train시에 당연히 잘못된 pseudo-label을 써서 EMA update를 하기 때문에 teacher network가 더 많은 OOD를 ID로 잘못 예측하게 된다. fig 2b처럼 모델이 오래 훈련될수록 OOD의 FPR이 증가한다.   

 이러한 문제를 해결하기 위해 본 논문에서는 detrimental effect를 제거하는 통합 프레임워크를 제안한다. instance-level task(e.g., object detection)에서 OOD detection을 더 잘하기 위해서 online/offline OOD detectior를 모두 고려했다. 
![image](https://user-images.githubusercontent.com/67910856/233138747-71654964-876e-4473-af9f-3d2e40b48c7b.png)   

Online/offline을 비교한 결과, online OOD detector는 pseudo-labeling의 불안정성(classification noise) 으로 인해 성능을 저하시키고 OOD detection, object detection 두 task를 shared architecture로 설계하게 되면 sub-optimal performance를 보였다. 따라서 본 논문은 Offline OOD detection 방법론을 사용할 것을 제안한다. 여기서 object detector와 OOD detector가 서로 구조상 분리되고 따로 학습된다.    
이 구조의 장점은 1) 기준 SSOD 방법론들과 호환되며 2) OOD detector과 obejct detectin 간의 competing task objectives를 완화한다는 점이다.    

그렇다면 어떤 offline OOD detector를 쓸까?
간단하게 pretrained ViT-B를 fine-tuning 할 수 있겠지만, 이 경우 ImageNet-21k로 pretraining을 해야 하고 large amount of annotated label이 필요하기 때문에 semi-supervised setting에 적절하지 않고 불공정한 비교를 초래할 수 있다. 따라서 OOD detector로  label annotation도 전혀 없어도 되며 ImageNet-1k 로만 학습되는 Self supervised DINO를 쓴다. multi-class(K+1) training으로 DINO를 fine tuning 한 다음 OOD score를 계산한다.    
 또 이전연구들과 달리 학습에 더 많은 OOD 데이터가 있을수록 좋다고 가정하지 않는다. 따라서 사용가능한 ID labeled data를 학습에 쓰고 background proposal patches를 OOD sample로 간주할 것을 제안한다. 구체적으로는 ground-truth box에 따라 이미지 패치를 자르고, 배경 패치는 ground-truth boxes 대비 IoU가 낮은 proposal boxes로부터 랜덤으로 샘플링해온다. Loss는 ID 패치를 해당하는 ID foreground label로 분류하도록, 배경 패치는 OOD class로 분류하도록 cross entropy loss를 계산한다. background instance를 OOD로 사용하는것은 sub-optimal이지만 OOD가 없을때 충분한 negative grad를 보인다고 주장한다.    

DINO model을 fine-tuning 한 후 DINO층을 얼려놓고 OOD detection을 수행한다. 
### 1. Mahalanobis Distance
![image](https://user-images.githubusercontent.com/67910856/233142621-cd76ca0c-b783-468f-9be6-92b147ab1b2e.png)
(2)식에서 f(x)는 패치 x를 통과시켜 DINO에서 뽑은 intermediate feature vector를 의미한다. ˜μk는 k class에 대해 추정된 평균 벡터이고  ˜Σ는 공분산 행렬이다. 이 평균과 공분산은 모두 available labeled data로부터 추정된 값이다 

### 2. Inverse Abstaining Confidence
![image](https://user-images.githubusercontent.com/67910856/233143006-3140d1ea-3ee7-477c-ba54-8cf9b1d0723e.png)
˜pK+1는 DINO classifier에서 K+1 th  class의 prediction confidence이다.   

### 3. Energy Score
![image](https://user-images.githubusercontent.com/67910856/233143377-673a1352-3979-4bec-860a-65acd511bf16.png)
(4)에서 T는 temperature value, k는 class 수, fi(x)는 logit을 의미한다. Energy loss 말고도 shannon entropy나 Euclidean Distance도 구려했는데 Energy가 제일 좋았다. threshold를 정해놓고 이것보다 낮으면 OOD sample로 간주했다. 

### Integration with Semi-Supervised Object Detection
pseudo label에서 OOD object를 필터링하는 것을 목표로 fig4처럼 confidence threshold를 거친 이후에 OOD filtering을 추가해서 trained OOD detector를 SSOD 방법과 통합한다. 즉, pseudo-boxes는 Teacher모델의 predicted boxes에 confidence threshold, OOD filtering을 순차적으로 적용한 다음 (3)으로 student 모델을 학습한다.
